library/src/main/java/com/novoda/downloadmanager/lib/DownloadTask.java

```diff
@@ -279,6 +280,10 @@ private void runInternal() {
 
             finalizeDestinationFile(state);
             finalStatus = DownloadStatus.SUCCESS;
+
+            if (batchHasPausedFiles(originalDownloadBatch.getBatchId())) {
+                finalStatus = DownloadStatus.PAUSED_BY_APP;
```

Comment:
You first set `finalStatus` as SUCCESS and then have an `if` to change it to `PAUSED_BY_THE_APP`. How about having it in the if - else manner ?